icp/src/python/triton_distributed/icp/event_plane.py

```diff
@@ -0,0 +1,154 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import dataclasses
+import json
+import re
+import uuid
+from abc import abstractmethod
+from datetime import datetime
+from typing import Any, AsyncIterator, Awaitable, Callable, List, Optional, Tuple, Union
+
+
+def _validate_subjects(subjects: List[str]) -> bool:
+    """
+    Checks if all strings in the list are alphanumeric and can contain underscores (_) and hyphens (-).
+
+    :param subjects: List of strings to validate
+    :return: True if all strings are valid, False otherwise
+    """
+    pattern = re.compile(r"^[a-zA-Z0-9_-]+$")
+
+    return all(pattern.match(subject) for subject in subjects)
+
+
+@dataclasses.dataclass
+class Topic:
```

Comment:
```suggestion
class EventTopic:
```
for consistency with other objects